Dok 123412

mystiberry-okta opened the pull request:
## Description:
- Improve ID token info in oauth doc.

### Resolves:

* [OKTA-123412](https://oktainc.atlassian.net/browse/OKTA-123412)

mystiberry-okta commented:
@johngronberg-okta  for the review.

johngronberg-okta commented on _source/_docs/api/resources/oidc.md:
```diff
@@ -734,22 +734,22 @@ of the response.
+| display               | How to display the authentication and consent UI. Valid values: *page* or *popup*.                                                                                                                                                                                                                                                                                                    ...
+| max_age               | Allowable elapsed time, in seconds, since the last time the end user was actively authenticated by Okta.                                                                                                                                                                                                                                                                              ...
+| response_mode         | How the authorization response should be returned.     [Valid values: *fragment*, *form_post*, *query* or *okta_post_message*](#parameter-details). If *id_token* or *token* is specified as the response type, then *query* isn't allowed as a response mode. Defaults to *fragment* in implicit and hybrid flow. Defaults to *query* in authorization code flow and cannot be set a ...
+| scope                 | `openid` is **required.** Other [scopes](/docs/api/resources/oidc.html#scope) may also be included.                                                                                                                                                                                                                                                                                   ...
```
the anchor should be for #scopes

mystiberry-okta replied:
ack

johngronberg-okta commented on _source/_standards/OAuth/index.md:
```diff
@@ -202,7 +202,7 @@ The payload includes the following reserved claims:
 ### Custom scopes and claims
 
-The admin can configure custom scopes and claims for the Authorization Server.
+The admin can configure custom scopes and claims for the Custom Authorization Server.
```
...and a groups claim for the Okta Authorization Server

do you want to call that out here?

mystiberry-okta replied:
Good to do so I think.

johngronberg-okta commented on _source/_standards/OAuth/index.md:
```diff
@@ -212,7 +212,7 @@ If the request that generates the access token contains any custom scopes, those
 #### Custom claims
 
-Custom claims are associated with scopes. If one of the associated scopes is granted to the Access Token, the custom claim is added into it. The value of a custom claim can be either an [expression](/reference/okta_expression_language/) or a group filter. The expression is evaluated at runtime, and if the evaluated result is null, that custom claim isn't added into the Access Token. The datatype  ...
+Custom claims are associated with scopes. If one of the associated scopes is granted to the access token, the custom claim is added into it. The value of a custom claim can be either an [expression](/reference/okta_expression_language/) or a group filter. The expression is evaluated at runtime, and if the evaluated result is null, that custom claim isn't added into the Access Token. The datatype  ...
```
custom claims can also be in id tokens.

mystiberry-okta replied:
Can we chat about this? Our doc is very vague about which claims show up in which tokens, which should probably be in this table: https://developer.okta.com/docs/api/resources/oauth2.html#response-parameters-1 

Also, we don't doc what you can't do if you don't have API AM enabled (in prod). Since API AM is a feature they must enable in prod, we have to talk about it.

johngronberg-okta commented on _source/_standards/OAuth/index.md:
```diff
@@ -273,16 +272,24 @@ Refresh Tokens can be revoked explicitly by making a [Revocation Request](/docs/
 
-The same validation steps for [OpenID Connect](/docs/api/resources/oidc.html#validating-id-tokens) can also be applied to ID Token for
-OAuth2, except the public keys should be retrieved via the [Get Keys endpoint](/docs/api/resources/oauth2.html#get-keys).
+The same validation steps for [OpenID Connect with the Okta Authorization Server](/docs/api/resources/oidc.html#validating-id-tokens) can also be applied to ID Token for
```
"... can also be applied to ID Tokens for..."

mystiberry-okta replied:
ack

johngronberg-okta commented on _source/_standards/OAuth/index.md:
```diff
@@ -273,16 +272,24 @@ Refresh Tokens can be revoked explicitly by making a [Revocation Request](/docs/
+
+## Requesting a Token
+
+For a list of tokens returned, depending on grant type and scope in your request, see [Response Parameters](/docs/api/resources/oidc.html#response-parameters-4).
```
should this link to an OAuth page?

mystiberry-okta replied:
Do you mean "for more about OAuth 2.0 and OpenID Connect tokens, see [specs]? 
Or, for more about tokens, see https://developer.okta.com/standards/OAuth/index.html#access-token (and ID Token)? Definitely should do one of those.

johngronberg-okta commented on _source/_standards/OAuth/index.md:
```diff
@@ -273,16 +272,24 @@ Refresh Tokens can be revoked explicitly by making a [Revocation Request](/docs/
+A Custom Authorization Server can issue an ID Token to the client, as the Okta Authorization Server can, but with the following differences:
+
+* The ID Token can contain [custom scopes](#custom-scopes) and [custom claims](#custom-claims).
+* The ID Token can't contain a reserved scope or claim called 'groups'. To obtain a claim with group information, administrators must define a custom claim with a group filter and associate it with a scope.
```
I don't think this is an actual reserved scope...

mystiberry-okta replied:
So, it should probably be 1. a reserved scope or 2. a claim called `groups`. We have this verbiage elsewhere, I'll check it.

mystiberry-okta commented:
I'm failing to be able to execute the simplest request token. I *think* I'm being forced to mac postman/no inteceptor, regardless of what I request. Working on it.

mystiberry-okta commented:
I've addressed all John's comments, and moved all these changes to https://github.com/okta/okta.github.io/pull/1092 to avoid clashes. Closing this PR. FYI @johngronberg-okta
